client/config.c

```diff
@@ -210,10 +244,45 @@ TDNFConfigFromCnfTree(PTDNF_CONF pConf, struct cnfnode *cn_top)
         {
             SET_STRING(pConf->pszPluginPath, cn->value);
         }
+        else if  (strcmp(cn->name, TDNF_CONF_KEY_TSFLAGS) == 0)
+        {
+            if (cn->value == NULL || strcmp(cn->value, "") == 0) {
+                pConf->rpmTransFlags = RPMTRANS_FLAG_NONE;
+            } else {
+                char *value = strdup(cn->value);
+                char *saveptr = NULL, *str, *token;
+                int i;
+
+                for (str = value; ; str = NULL){
+                    token = strtok_r(str, " ", &saveptr);
+                    if (token == NULL)
+                        break;
+
+                    for (i = 0; rpmtransflags_map[i].name != NULL; i++) {
+                        if (strcmp(token, rpmtransflags_map[i].name) == 0){
+                            pConf->rpmTransFlags |= rpmtransflags_map[i].flag;
```

Comment:
should `rpmTransFlags` should be set to zero before entering loop?

Reply:
No, it should definitely not. This function may be called multiple times - first for the config which may set flags, and then again for the `--setopt` options, which may set additional flags without resetting them. Only if we explicirely set them to "" should they be reset. See https://dnf.readthedocs.io/en/latest/command_ref.html:

"Values for configuration options like excludepkgs, includepkgs, installonlypkgs and tsflags are appended to the original value, they do not override it. However, specifying an empty value (e.g. --setopt=tsflags=) will clear the option."